### ISSUE TITLE
Differences in the libraries' names

### ISSUE THREAD

**Issue by daniilpetrov** [kcat/openal-soft#491]
Why does MinGW build a library with names libOpenAL32.dll.a and OpenAL32.dll, and an executable file, which requires them, asks for soft_oal.dll?

**Comment by kcat** [comment]
Which executable file asks for `soft_oal.dll`?

`libOpenAL32.dll.a` and `OpenAL32.dll` is what applications use directly. You link to `libOpenAL32.dll.a` and the resulting executable uses`OpenAL32.dll`. When you build OpenAL Soft normally, this is what you get and all that's needed to link and run apps with OpenAL Soft. Anything more is entirely optional.

For a full(er) explanation, OpenAL was designed to be implemented by multiple hardware vendors, back when hardware audio was still a thing. The `OpenAL32.dll` that apps link to at runtime could instead be a "router" DLL, and this router would look for various OpenAL "drivers" on the system and transparently present them to the app through a unified interface. `ct_oal.dll` would provide hardware devices for Creative's sound cards, `wrap_oal.dll` would provide `Generic Hardware`/`Software` devices for compatibility fallback, etc. `soft_oal.dll` is what OpenAL Soft would be named to be seen by this router, and would provide `OpenAL Soft` devices with its capabilities and feature-set.

OpenAL Soft can be configured to build its own router, so `OpenAL32.dll` is a router and `soft_oal.dll` is the main library. Your app would link to `libOpenAL32.dll.a`, and you'd provide both `OpenAL32.dll` (the router) and `soft_oal.dll` (the core library) with the application. But this is not needed if you don't need to support multiple drivers.

**Comment by daniilpetrov** [comment]
No :) I`m link to libOpenAL32.dll.a and the resulting executable uses soft_oal.dll ))) Compilator is GCC with Mingw-w64.

**Comment by kcat** [comment]
How exactly do you build and link your app? Linking to `libOpenAL32.dll.a` will make the resulting executable implicitly depend on `OpenAL32.dll`. With the default configuration, `OpenAL32.dll` is the main library and `soft_oal.dll` won't be built or exist anywhere, so your app shouldn't be able to link to the latter DLL.

**Comment by daniilpetrov** [comment]
Using GUI CMake, GCC and Wingw-w64, I build Soft OpenAL by default, I only specify the build configuration - Release. Then, when compiling, I link libOpenAL32.dll.a and the final executable file requires the soft_oal.dll library. Even if I link libOpenAL32.dll, the final executable file still requires soft_oal.dll))) To build Windows binaries for Linux using CMake, I use this script: https://gist.github.com/peterspackman/8cf73f7f12ba270aa8192d6911972fe8

**Comment by kcat** [comment]
> I link libOpenAL32.dll.a and the final executable file requires the soft_oal.dll library.

I don't see any reason for that. The `soft_oal` name isn't used at all if not building the router or an OSX framework, and the compiler creates the `libOpenAL32.dll.a` import library when creating `OpenAL32.dll`, so there should be no expectation for `soft_oal.dll` with apps that link to it. Are you sure you're linking to the newly-created `libOpenAL32.dll.a`, and its using the newly-created DLL?

**Comment by daniilpetrov** [comment]
Certainly! And also the function names are not imported from the file libOpenAL32.dll.a, which goes with your openal-soft-1.21.0-bin.zip, I have to import them from the file soft_oal.dll, from the same place. Do you build binaries in MSYS2 or Mingw-w64?

**Comment by kcat** [comment]
They're built with MinGW-w64, cross-compiled on Linux. The import library has no reference to `soft_oal.dll`:
```
$ strings libOpenAL32.dll.a | grep -i '\.dll'
OpenAL32.dll
```
So there's no reason it should be trying to load symbols from `soft_oal.dll`.
